### DISCUSSION TITLE
tlg-catalog/stable/tables/demography/dmt01

### DISCUSSION THREAD

**Discussion by giscus** [insightsengineering/tlg-catalog#247] [General]
# tlg-catalog/stable/tables/demography/dmt01

Demographics and Baseline Characteristics

https://insightsengineering.github.io/tlg-catalog/stable/tables/demography/dmt01.html

**Comment by liang2022** [comment]
How to change SD in the 'age' column to two decimal places, while keeping 'mean' to one decimal place

**Reply by edelarua** [reply]
Hi @liang2022,

This format is not currently included as a supported format within `formatters`, but can be applied using a custom `sprintf` formatting function. This is applied to the DMT01 example code within the `analyze_vars` call of the table layout as follows:

```r
result <- basic_table(show_colcounts = TRUE) %>%
  split_cols_by(var = "ACTARM") %>%
  add_overall_col("All Patients") %>%
  analyze_vars(
    vars = vars,
    var_labels = var_labels,
    .formats = c(mean_sd = sprintf_format("%.1f (%.2f)"))
  ) %>%
  build_table(adsl)
```

Please note that this will change the output format for all numerical variables summarized in your table, not only `AGE`.

**Reply by liang2022** [reply]
Hi edelarua,
Thanks for your help. if I want age and bmi to retain different numbers of decimal places, how should I operate? At the same time, I want to hide the 'n' in gender

**Reply by edelarua** [reply]
@liang2022 in that case, it's easiest to use consecutive calls of `analyze_vars`, with different settings specified where needed.

For example, with `AGE` and `SEX` removed from `vars` and `var_labels`:

```r
result <- basic_table(show_colcounts = TRUE) %>%
  split_cols_by(var = "ACTARM") %>%
  add_overall_col("All Patients") %>%
  analyze_vars(
    vars = "AGE",
    var_labels = "Age (yr)",
    .formats = c(mean_sd = sprintf_format("%.1f (%.2f)"))
  ) %>%
  analyze_vars(
    vars = "SEX",
    var_labels = "Sex",
    .stats = "count_fraction"
  ) %>%
  analyze_vars(
    vars = vars,
    var_labels = var_labels
  ) %>%
  build_table(adsl)
```

**Comment by DSuser84** [comment]
Hi @edelarua ,
Related to the above question, do you have any tricks for specifying custom decimal places so that they will be outputted appropriately in the R-Shiny UI?  
My apologies if you implicitly answered this above. However, I am unsure how this would be implemented into the R-Shiny app.
I would really appreciate any help you can provide.

**Reply by edelarua** [reply]
Hi @DSuser84,

Unfortunately at this time there is no option to change decimal place settings in the corresponding teal app. If necessary, you could create a custom teal module to do this by copying the code for the [`tm_t_summary()`](https://insightsengineering.github.io/teal.modules.clinical/latest-tag/reference/tm_t_summary.html) module from the {teal.modules.clinical} package and editing the table code as discussed above.

**Reply by DSuser84** [reply]
Thanks @edelarua, also thank you for your work in general on this software. The R Shiny UI and the online documentation are amazing.
